### PR TITLE
Add the Array.sliding & List.sliding functions

### PR DESCRIPTION
**Pull request by Dean177**

### CODE REVIEW

File: bs/src/tablecloth.ml [hunk 1]
```diff
@@ -174,6 +184,10 @@ module Array = struct
 end
 
 module List = struct
+  type 'a t = 'a list
+
+  let isEmpty t = match t with [] -> true | _ -> false
```
Comment (pbiggar) [inline bs/src/tablecloth.ml]: This could be `t <> []`

File: native/src/tablecloth.mli [hunk 2]
```diff
@@ -277,8 +277,28 @@ Array.append fourtyTwos eightyOnes = [|42; 42; 81; 81; 81|];]} *)
   
     {[Array.slice ~from:(-2)  ~to_:(-1) [|0; 1; 2; 3; 4|] = [|3|]]} *)
 
+  val sliding : ?step:int -> 'a t -> size:int -> 'a t t
+  (** Provides a sliding 'window' of sub-arrays over an array.
+
+    The first sub-array starts at index [0] of the array and takes the first [size] elements.
+
+    The sub-array then advances the index [step] (which defaults to 1) positions before taling the next [size] elements.
```
Comment (pbiggar) [inline native/src/tablecloth.mli]: "taking"

File: native/test/test.ml [hunk 3]
```diff
@@ -59,21 +59,25 @@ let t_Array () =
     "map2 one element"
     (List.map2 ~f:( + ) [ 1 ] [ 1 ])
     [ 2 ] ;
+
```
Comment (pbiggar) [inline native/test/test.ml]: Odd formatting change. Was it correctly formatted before, or is it correctly formatted now?

Comment (Dean177) [reply native/test/test.ml]: This was me sorry. `ocamlformat` won't remove excess whitespace, but it doesn't mind the statements being right net to each other.

**Pull request merged**